File: 0205-isomorphic-strings/0205-isomorphic-strings.cs
using System.Collections.Generic;

public class Solution {
    public bool IsIsomorphic(string s, string t) {
        if (s.Length != t.Length)
            return false;

        Dictionary<char, char> charMappingMap = new Dictionary<char, char>();

        for (int i = 0; i < s.Length; i++) {
            char original = s[i];
            char replacement = t[i];

            if (!charMappingMap.ContainsKey(original)) {
                if (!charMappingMap.ContainsValue(replacement))
                    charMappingMap.Add(original, replacement);
                else
                    return false;
            } else {
                char mappedCharacter = charMappingMap[original];
                if (mappedCharacter != replacement)
                    return false;
            }
        }

        return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.js
function isIsomorphic(s, t) {
    if (s.length !== t.length) {
        return false;
    }

    const charMappingMap = new Map();

    for (let i = 0; i < s.length; i++) {
        const original = s[i];
        const replacement = t[i];

        if (!charMappingMap.has(original)) {
            if (![...charMappingMap.values()].includes(replacement))
                charMappingMap.set(original, replacement);
            else
                return false;
        } else {
            const mappedCharacter = charMappingMap.get(original);
            if (mappedCharacter !== replacement)
                return false;
        }
    }

    return true;
}
